src/librustc_trait_selection/traits/wf.rs

```diff
@@ -34,6 +34,22 @@ pub fn obligations<'a, 'tcx>(
     }
 }
 
+/// Returns the set of obligations needed to make `c` well-formed.
+pub fn const_obligations<'a, 'tcx>(
+    infcx: &InferCtxt<'a, 'tcx>,
+    param_env: ty::ParamEnv<'tcx>,
+    body_id: hir::HirId,
+    constant: &ty::Const<'tcx>,
+    span: Span,
+) -> Vec<traits::PredicateObligation<'tcx>> {
```

Comment:
The caveat about the return type applies: it should be the same as `obligations`.

Reply:
ayy, will update compute_const to actually return if there was progress